### ISSUE TITLE
sql/multitenancy: eventlog should use sql instance IDs to be compatible with multitenancy

### ISSUE THREAD

**Issue by asubiotto** [cockroachdb/cockroach#50047]
Seen in `testdata/logic_test/event_log`:
```
        testdata/logic_test/event_log:22: SELECT "reportingID", info::JSONB->>'TableName'
          FROM system.eventlog
         WHERE "eventType" = 'create_table'
        expected:
            1  test.public.a
            1  test.public.b

        but found (query options: "rowsort" -> ignore the following ordering of rows) :
            10001  test.public.a
            10001  test.public.b
```
The node ID is stored and printed out (the 10001 is a bogus node ID). These should be sql instance IDs instead (#48009)

Jira issue: CRDB-4158

**Comment by vy-ton** [comment]
cc @thtruo for awareness of log issue with Serverless

**Comment by knz** [comment]
This has been addressed in #64894.
